### DISCUSSION TITLE
Table row button - Action to update field value not working

### DISCUSSION THREAD

**Discussion by bhotrock** [Budibase/budibase#7033] [Help] [answered]
I'm very new to Budibase.  I'm sure I've missed a step along the way somewhere.  Here's what I've done:
On the Data side, I am using the Budibase DB.  I created a table with four fields and two rows, for testing.  I have one field that is a boolean checkbox.  I want to be able to press a button at the start of a row and change the value of the checkbox from unchecked to checked.

In the design, I have my DataProvider, and a Table.  With the Table selected, I added the Button component which placed a button at the start of each row.  This is what I want.

I attempt to set the "on-click" action for the button to:  Update Field Value.  In the settings I do not have a drop-down option for "Form", so I leave it alone.  (I have not defined any forms.)  "Type" is set to "set value".  There are no drop-down options for "Field" so I type in the field name of the checkbox field.  In "Value", I've tried the following:  true, false, 0, 1.  Save and publish.  

When I test, the button appears to do nothing.  I'm not sure if it's a problem with the Data, Design, or possibly permissions.  Can anyone tell me what I'm missing? 

Thank you!!!
![Data](https://user-images.githubusercontent.com/19352294/182245532-581bbedf-d6cf-4c0d-b571-1a729965310e.png)

![Check-in Screen](https://user-images.githubusercontent.com/19352294/182245167-7d7ca3cb-6ca7-4fca-aa17-987d9132dff6.png)
![Action](https://user-images.githubusercontent.com/19352294/182245820-d964c30d-2365-4d34-82d6-80a2b1a90594.png)

**Comment by R2bEEaton** [comment, marked as answer]
Hello and welcome!

I can see why you would use "Update Field" - it makes total sense grammatically. However, "Update Field" is reserved for Form Fields.

### Save Row

What you're looking for is the "Save Row" action. The [documentation](https://docs.budibase.com/docs/actions) for it says that if you save row and include the `_id` field of an existing row, it will update that row. Make sure to modify the `{{ New Table.Histable._id }}` to include the correct names of your elements. You can click the lightning bolt and use the Handlebar search to search for `_id` and you should find the correct path.

![image](https://user-images.githubusercontent.com/34921506/182832565-b0d6a3fc-80b3-4ba4-8066-446de935b7a0.png)

### Bonus tip!
Let's say the person already arrived. Now there's no need to display the "Arrived" button. Use "Conditions" to hide it when `{{ New Table.Histable.Arrived }}` (or your equivalent Handlebar) is True.

![image](https://user-images.githubusercontent.com/34921506/182833031-cd5b5002-c62a-4739-8bba-61e78d2ecc15.png)

You can even add a second button and have the opposite condition to provide functionality for *un-arriving* a user, depending on what your use-case is.

### Enjoy!

**Reply by bhotrock** [reply]
R2bEEaton, Thank you SO much.  That makes sense and solved my issue.  And thanks for the bonus tip.  That's a nice little feature that will help.  Have a great day!
